lib/marginalia/comment.rb

```diff
@@ -4,11 +4,16 @@ module Marginalia
   module Comment
     mattr_accessor :components, :lines_to_ignore
     Marginalia::Comment.components ||= [:application, :controller, :action]
+    THREAD_LOCAL_KEY_FOR_JOB = 'marginalia.comment.job'
 
     def self.update!(controller = nil)
       @controller = controller
     end
 
+    def self.update_job!(job)
+      Thread.current[THREAD_LOCAL_KEY_FOR_JOB] = job
```

Comment:
Should this be made thread-safe if `@controller` is already being stored in a non-thread-safe way?  It should probably be consistent.